lib/acme/client/faraday_middleware.rb

```diff
@@ -35,6 +37,12 @@ def on_complete(env)
 
   private
 
+  def jws_header
+    headers = { nonce: pop_nonce, url: env.url.to_s }
+    headers[:kid] = client.kid if @mode == :kid
+    headers
```

Comment:
Style preference, but you could write this as:

```ruby
{ nonce: pop_nonce, url: env.url.to_s }.tap do |headers|
  headers[:kid] = client.kid if @mode == :kid
end
```

Reply:
I think its a bit more readable to just return the variable